jenkins/databricks/test.sh

```diff
@@ -89,21 +89,20 @@ run_pyarrow_tests() {
         bash integration_tests/run_pyspark_from_build.sh -m pyarrow_test --pyarrow_test --runtime_env="databricks" --test_type=$TEST_TYPE
 }
 
-## limit parallelism to avoid OOM kill
-export TEST_PARALLEL=${TEST_PARALLEL:-4}
-
-if [[ $TEST_MODE == "DEFAULT" ]]; then
+## Separate the integration tests into two parts to speed up the testing process. Run each part in parallel on separate Databricks clusters.
+IT_PART=${IT_PART:-"part1"}
+if [[ $TEST_MODE == "DEFAULT" && $IT_PART == "part1" ]]; then
     bash integration_tests/run_pyspark_from_build.sh --runtime_env="databricks" --test_type=$TEST_TYPE
+fi
 
+## Run tests with jars building from the spark-rapids source code
+if [[ "$(pwd)" == "$SOURCE_PATH" && $IT_PART == "part2" ]]; then
```

Comment:
Seems like this will cause the below tests to not run when a developer runs jenkins/databricks/test.sh since IT_PART will not be part2 here?  Wondering if we need a separate TEST_MODE for parallel testing that isn't DEFAULT so we can distinguish between runs by developers and runs by CI in parallel.

Reply:
Good catch,

I agree we can add new options like `TEST_MODE=part1 or `TEST_MODE=part2` to parallel for CI,

and for developers, TEST_MODE=DEFAULT still runs all tests

Reply:
That sounds better